File: Container With Most Water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l, r = 0, len(height) - 1
        max_area = 0

        while l < r:
            diff = r - l
            h = min(height[l], height[r])
            curr = diff * h
            max_area = max(max_area, curr)

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return max_area

File: Container With Most Water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size()-1;
        int maxarea = 0;
        while (l < r) {
            int diff = r - l;
            int h = min(height[l], height[r]);
            int curr = diff * h;
            maxarea = max(maxarea, curr);
            if (height[l] < height[r])
                l++;
            else
                r--;
        }
        return maxarea;
    }
};